test/project_types/script/layers/infrastructure/metadata_repository_test.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "project_types/script/test_helper"
+
+describe Script::Layers::Infrastructure::MetadataRepository do
+  let(:instance) { Script::Layers::Infrastructure::MetadataRepository.new(ctx: ctx) }
+  let(:ctx) { TestHelpers::FakeContext.new }
+
+  describe ".get_metadata" do
+    subject { instance.get_metadata }
+
+    describe "when metadata file is present and valid" do
+      let(:metadata_file_location) { "metadata.json" }
+      let(:metadata_json) do
+        JSON.dump(
+          {
+            schemaVersions: {
+              example: { major: "1", minor: "0" },
+            },
+          },
+        )
+      end
+
+      it "should return a proper metadata object" do
+        File.expects(:read).with(metadata_file_location).once.returns(metadata_json)
```

Comment:
I'd recommend using FakeFS here and actually writing this content to the file in a `before` block. Using a test double helps us suppress more FS operations sneaking into our tests which makes our tests more readable and easier to refactor later!